src/index.ts

```diff
@@ -136,11 +143,22 @@ export default function collectFs(
       const target = join(destination, rel);
       await ensureDir(dirname(target));
       await promises.copyFile(path, target, constants.COPYFILE_FICLONE);
+
+      if (onEvent) {
+        onEvent({
+          type: eventType,
+          payload: { filesStackCount: Object.keys(files).length, rel },
```

Comment:
Calling `keys` every time is not very performant, as it will create an array each time (it's a loop we can avoid). instead, use a `let counter` which will be way faster

Reply:
But since we have just some hundred elements in the array, the ´Object.keys´ is a better choice.

Reply:
you are really looking for the complicated stuff all the time. NO LOOPS. every time a property is added to the object, sum one to a counter, then read it instead of looping through the object

Reply:
That solution would add way more complexity, than the pros. 
I would suggest to refactor the  ´files´ object to a ´new Map´, and that could add a real value.